tests/pytests/test_expire.py

```diff
@@ -218,3 +220,89 @@ def expireDocs(env, isSortable, expected_results, isJson):
+# The field_to_schema_list allows specifying additional schema keywords for the field
+# e.g SORTABLE, this can affect the expected results
+# The value for each field will be 't' if the field is marked to expire and the document is marked to expire,
+# otherwise 'f'
+def testExpireFields(env, field_name_to_expire, document_name_to_expire, field_to_schema_list:dict[str, list[str]]={}
+                     , expected_results=[]):
+    conn = getConnectionByEnv(env)
+    conn.execute_command('DEBUG', 'SET-ACTIVE-EXPIRE', '0')
+    fields_to_expire = {}
+    schema = []
+    for field, expire in field_name_to_expire.items():
+        schema.append(field)
+        for keyword in field_to_schema_list.get(field, []):
+            schema.append(keyword)
+        schema.append('TEXT')
+        if expire is not None:
+            if expire not in fields_to_expire:
+                fields_to_expire[expire] = []
+            fields_to_expire[expire].append(field)
+    conn.execute_command('FT.CREATE', 'idx', 'SCHEMA', *schema)
+
+    documents = []
+    for document_name, will_expire in document_name_to_expire.items():
+        expire_text_list = ['t' if (b and will_expire) else 'f' for b in field_name_to_expire.values()]
+        values = list(chain(*zip(field_name_to_expire.keys(), expire_text_list)))
+        conn.execute_command('HSET', document_name, *values)
+        documents.extend([document_name, values])
+
+    expected_before_expiration = [len(document_name_to_expire), *documents]
+    env.expect('FT.SEARCH', 'idx', '*').equal(expected_before_expiration)
+
+    for doc_name, expire in document_name_to_expire.items():
+        if not expire:
+            continue
+        for duration, fields in fields_to_expire.items():
+            conn.execute_command('HPEXPIRE', doc_name, '1', 'FIELDS', str(len(fields)), *fields)
+    # now allow active expiration to delete the expired fields
+    conn.execute_command('DEBUG', 'SET-ACTIVE-EXPIRE', '1')
+    time.sleep(0.5)
+    env.expect('FT.SEARCH', 'idx', '*').equal(expected_results)
+    conn.execute_command('FLUSHALL')
+
+
+def testSingleExpireField(env):
+    testExpireFields(env, field_name_to_expire={'x': 1},
+                     document_name_to_expire={'doc1': True, 'doc2': False},
+                     expected_results=[1, 'doc2', ['x', 'f']])
+
+
+def testTwoFieldsOneOfThemWillExpire(env):
+    testExpireFields(env, field_name_to_expire={'x': None, 'y': 1},
+                     document_name_to_expire={'doc1': True, 'doc2': False},
+                     expected_results=[2, 'doc1', ['x', 'f'], 'doc2', ['x', 'f', 'y', 'f']])
+
+
+def testSingleSortableFieldWithExpiration(env):
+    testExpireFields(env, field_name_to_expire={'x': 1},
+                     document_name_to_expire={'doc1': True, 'doc2': False},
+                     field_to_schema_list={'x': ['SORTABLE']},
+                     expected_results=[2, 'doc1', ['x', 't'], 'doc2', ['x', 'f']])
```

Comment:
How is this results returned? `doc1` should be deleted.
Please validate this test.